Initial inheritance Repo parent class

foster55f opened the pull request:
What's this PR do?
This PR update created the initial file for our new Repo Parent Class.  I added the data ,id, and user properties which were included in all of the Sleep,Activity, and Hydration classes.  I also added a `getFitData` method which will end up removing the repetitive method included in all 3 child classes.  I have only updated this for the Sleep child class currently.  This PR creates the Repo parent class, testing for the first method, and updates to the Sleep class for all Sleep properties and new variable names changed in Sleep class methods.  I have also removed the Sleep getUserData method since it is now being called in parent class.

Where should the reviewer start?
I would take a look at the Repo properties and Sleep properties to see the repetition and how parent class is able to assist child with providing those properties.  Also the `getFitData` method will now take care of the repetitive `getUserData` methods included in Sleep,Activity,and Hydration.

How should this be manually tested?
I would pull the repo down and console.log to make sure data is being pulled correctly.  All tests passed on my end.

Any background context you want to provide?
This Repo class will continue to be built out to provide more methods for the repetitive methods in child classes.  
@brittanystoroz 
@khalidwilliams

nicknist commented:
Hey Foster! 

This looks so great! I pulled the repo down and all my tests passed as well (picture of tests below). I'm impressed with how quickly you implemented inheritance and did it so well. Two pieces of feedback: 

1. I did notice that there's a console.log in the Repo-test.js file on line 64. It was very helpful to make sure the data is correct and passing through the correct thing, but we'll want to take that out sometime this week along with the commented out code. 

2. This is the first time I noticed the 'See if the tests are running', which is coming from a sample-test.js file. We can probably take that out (probably inherited from the old repo). 
  
<img width="1550" alt="Screen Shot 2019-10-22 at 5 52 56 PM" src="https://user-images.githubusercontent.com/47511765/67344299-ea343b80-f4f4-11e9-8915-06d645b063d0.png">

Overall, looks great! GO TEAM :)

Alkiiro commented on src/index.js:
```diff
@@ -65,6 +66,7 @@ const hydrationRepository = new HydrationRepository(hydrationData, randomId);
 const sleepRepository = new SleepRepository(sleepData, randomId);
 const activityRepository = new ActivityRepository(randomId, activityData);
 const user = new User(userRepository.getUserData());
+console.log(sleepRepository )
```
Don't forget to take out those console.logs!

foster55f replied:
Cool thanks for checking.  Ill get those console.log removed, and the  `'See if the tests are running'` file.  Thanks team!

Alkiiro commented on test/Repo-test.js:
```diff
@@ -0,0 +1,87 @@
+describe('Repo', () => {
+
+    it('should be a function', () => {
+      console.log(repo.getFitData())
```
Unnecessary console.log.

Alkiiro commented on src/SleepRepository.js:
```diff
@@ -1,13 +1,16 @@
-  }
+  // getSleepData() {
+  //   return this.sleepData.filter(user => user.userID === this.id);
+  // }
```
Like you already said in person @foster55f, let's take these commented lines out.

foster55f commented:
I removed console.log from Repo test file and the commented out sections from Sleep class.  Please review last commit for this PR.  Thanks

foster55f commented:
I just added additional updates to this branch.  I updated the Activity class to include similar changes made to Sleep class.  I had to change order of the way data and id was passed into the Activity class to match up with parent and Sleep. I removed test for the `getFitData` in the activity test file and verified test in repo test file.  I also updated the new instantiation of  activity in index.js to fix how data and id were passed into the new activity instantiation.  Please review these additional changes.  Thanks
@nicknist 
@Alkiiro
